src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleObject.cs

```diff
@@ -72,11 +67,13 @@ private static bool WriteEndObject(ref WriteStack state)
             return true;
```

Comment:
Code path in line 62 is not really needed, it is only hit when root object finishes and just sets root's WriteStackFrame properties to null/false; which no longer matters at the end of serialization.
@steveharter

Reply:
Yes thanks. It is not needed - any future push() from a subsequent serialization will reset the state.